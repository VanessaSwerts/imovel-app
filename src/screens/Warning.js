import React from 'react'
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native'
import { FontAwesome } from '@expo/vector-icons'

import colors from '../constants/colors.json'

export default function Warning({ navigation }) {
  return (
    <View style={styles.container}>

      <FontAwesome name="warning" size={64} color={colors['blue-secondary']} />

      <Text style={styles.message} numberOfLines={2}> Ops, para prosseguir é necessário que você faça o login!</Text>
      <Text style={styles.secondaryMessage}> Caso não tenha uma conta, cadastra-se!</Text>

      <View style={styles.buttonContainer}>
        <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('SignIn')}>
          <Text style={styles.buttonText}>Login</Text>
        </TouchableOpacity>

        <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('SignUp')}>
          <Text style={styles.buttonText}>Cadastrar</Text>
        </TouchableOpacity>
      </View>


    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors["platinum"],
    justifyContent: 'center',
    alignItems: 'center',
    padding: 30,
  },

  message: {
    marginTop: 20,
    fontSize: 18,
    fontWeight: "700",
    color: colors['blue'],
    textAlign: "left",
  },

  secondaryMessage: {
    marginTop: 5,
    width: 300,
    fontSize: 14,
    fontWeight: "500",
    color: colors['blue'],
    textAlign: 'center',
  },

  buttonContainer: {
    marginTop: 20,
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },

  button: {
    marginHorizontal: 10,
    height: 35,
    width: 120,
    backgroundColor: colors['yellow'],
    borderRadius: 24,
    marginTop: 10,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
  },

  buttonText: {
    color: colors['blue'],
    fontWeight: 'bold',
    fontSize: 16
  }

})