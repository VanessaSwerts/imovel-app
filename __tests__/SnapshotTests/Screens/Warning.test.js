import React from 'react'
import { act, create } from 'react-test-renderer'
import Warning from '../../../src/screens/Warning'

jest.useFakeTimers()
jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon')

describe('<Warning />', () => {

  it('render correctly', () => {
    act(() => { tree = create(<Warning />) })

    expect(tree.toJSON()).toMatchSnapshot()
  })


})