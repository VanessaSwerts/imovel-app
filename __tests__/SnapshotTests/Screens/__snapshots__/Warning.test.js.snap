// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Warning /> render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E5E5E5",
      "flex": 1,
      "justifyContent": "center",
      "padding": 30,
    }
  }
>
  <Icon
    color="#19294d"
    name="warning"
    size={64}
  />
  <Text
    numberOfLines={2}
    style={
      Object {
        "color": "#14213D",
        "fontSize": 18,
        "fontWeight": "700",
        "marginTop": 20,
        "textAlign": "left",
      }
    }
  >
     Ops, para prosseguir é necessário que você faça o login!
  </Text>
  <Text
    style={
      Object {
        "color": "#14213D",
        "fontSize": 14,
        "fontWeight": "500",
        "marginTop": 5,
        "textAlign": "center",
        "width": 300,
      }
    }
  >
     Caso não tenha uma conta, cadastra-se!
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#FCA311",
          "borderRadius": 24,
          "height": 35,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginTop": 10,
          "opacity": 1,
          "width": 120,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#14213D",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#FCA311",
          "borderRadius": 24,
          "height": 35,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginTop": 10,
          "opacity": 1,
          "width": 120,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#14213D",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Cadastrar
      </Text>
    </View>
  </View>
</View>
`;
